mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1989,6 +1990,50 @@ func TestClientSideEncryptionProse(t *testing.T) {
 			}
 		})
 	})
+
+	mt.RunOpts("20. Bypass creating mongocryptd client when shared library is loaded",
+		noClientOpts, func(mt *mtest.T) {
+
+			cryptSharedLibPath := os.Getenv("CRYPT_SHARED_LIB_PATH")
+			if cryptSharedLibPath == "" {
+				mt.Skip("CRYPT_SHARED_LIB_PATH not set, skipping")
+				return
+			}
+
+			kmsProviders := map[string]map[string]interface{}{
+				"local": {
+					"key": localMasterKey,
+				},
+			}
+
+			// Listen for connections in a separate goroutine.
+			listener := listenForConnections(t, func(_ net.Conn) {
+				mt.Fatalf("Unexpected connection created to mock mongocryptd goroutine")
```

Comment:
Optional: Consider closing the connection in the listener callback.

```suggestion
			listener := listenForConnections(t, func(c net.Conn) {
				mt.Fatalf("Unexpected connection created to mock mongocryptd goroutine")
				c.Close()
```